src/src/SortedArray.hpp

```diff
@@ -58,14 +46,9 @@ class SortedArray : protected SortedArrayView<T, INDEX_TYPE>
 {
 public:
 
-  // Wrapper needs access to the data and resize methods, however these methods
-  // need to be private so Wrapper is a friend class.
-  template <class U>
-  friend class geosx::dataRepository::Wrapper;
-
   // These are needed by Wrapper.
   using value_type = T;
-  using pointer = T *;
+  using pointer = T const *;
```

Comment:
Why the const?

Reply:
`SortedArray::operator[]` returns `T const &`.